### PR TITLE
Supplier V2 Functionality

### PR DESCRIPTION
**Pull request by TylerGarlick**
- Added babel, eslint, other build tools
- Added supplier v2 functionality
- Refactor of routing and v1/v2
- Added 'current' as a pointer to v2

### CODE REVIEW

File: src/v2/availability.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      return http.put(uri, payload);
+    },
+
+    delete(pmcId, unitId) {
+      const uri = getRoute(Kind.COLLECTION, pmcId, unitId);
+      return http.delete(uri);
+    },
+
+    status(pmcId, unitId, workflowId) {
+      const uri = getRoute(Kind.WORKFLOW, pmcId, unitId, workflowId);
```
Comment (jkalexander2) [inline src/v2/availability.js]: I'm not sure if I missed something but in your routes file there is no mapping for Kind.WORKFLOW for the availability routes.

File: src/v2/shared/routes.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  pmcs: {
+    collection: template('/hub/v2/pmcs'),
+    id: template('/hub/v2/pmcs/{pmcId}'),
+    workflow: template('/hub/v2/pmcs/workflows/{workflowId}')
+  },
+
+  policies: {
+    collection: template('/hub/v2/pmcs/{pmcId}/policies'),
+    id: template('/hub/v2/pmcs/{pmcId}/policies/{policyId}'),
+    appliesTo: template('/hub/v2/pmcs/{pmcId}/policies/{policyId}/applies-to'),
```
Comment (jkalexander2) [inline src/v2/shared/routes.js:41]: Seems like appliesTo isn't being used. Was this implemented preemptively?

File: src/v2/stay-restrictions-rate-plans.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      return http.patch(uri, payload);
+    },
+
+    update(pmcId, unitId, ratePlanId, payload) {
+      const uri = getRoute(Kind.COLLECTION, pmcId, unitId, ratePlanId);
+      return http.put(uri, payload);
+    },
+
+    status(pmcId, workflowId) {
+      const uri = getRoute(Kind.WORKFLOW, pmcId, workflowId);
```
Comment (jkalexander2) [inline src/v2/stay-restrictions-rate-plans.js]: Seems like there is no Kind.WORKFLOW mapping for stayRestrictions.

File: src/v2/availability.js (hunk 1 above)
Comment (TylerGarlick) [inline src/v2/availability.js]: This must've been a refactor issue.  It should be something like this: 

``` js
export const Kind = {
  COLLECTION: 'collection',
  ID: 'id',
  APPLIES_TO: 'appliesTo',
  WORKFLOW: 'workflow',
  DATE: 'date',
  CHECK: 'check'
};
```

File: src/v2/shared/routes.js (hunk 2 above)
Comment (TylerGarlick) [inline src/v2/shared/routes.js:41]: Most of the routes work I got from here: https://public-api.apis.leisurelink.com/docs/hub#/

File: src/v2/stay-restrictions-rate-plans.js (hunk 3 above)
Comment (TylerGarlick) [inline src/v2/stay-restrictions-rate-plans.js]: Removed. Good catch

File: src/v2/availability.js (hunk 1 above)
Comment (jkalexander2) [inline src/v2/availability.js]: I saw that. I guess I didn't explain what I meant properly. Seems like there is no Kind.WORKFLOW mapping for availability, just like there wasn't one for stayRestrictions.

Comment (TylerGarlick) [inline src/v2/availability.js]: Gotcha, should be fixed now.

**Pull request merged**